lua/telescope/builtin/git.lua

```diff
@@ -107,37 +107,35 @@ end
 
 git.status = function(opts)
   local output = vim.split(utils.get_os_command_output('git status -s'), '\n')
-  local results = {}
-  for _, v in ipairs(output) do
-    if v ~= "" then
-      local mod, fname = string.match(v, '(..)%s(.+)')
-      if mod ~= 'A ' and mod ~= 'M ' and mod ~= 'R ' and mod ~= 'D ' then
-        table.insert(results, { mod = mod, file = fname })
-      end
-    end
-  end
 
-  if vim.tbl_isempty(results) then
+  if vim.tbl_isempty(output) then
     print('No changes found')
     return
   end
 
   pickers.new(opts, {
     prompt_title = 'Git Status',
     finder = finders.new_table {
-      results = results,
+      results = output,
       entry_maker = function(entry)
+        if entry == '' then return nil end
+        local mod, file = string.match(entry, '(..)%s(.+)')
+        if mod:sub(1, 1) == 'R' or mod:sub(1, 1) == 'C' then
+          _, file = string.match(file, '(.+)%s.+%s(.+)')
```

Comment:
Whats matched inside the `()` will be returned. Just not setting the first `()` lets you get rid of `_, `.

And i think we can do this in one regex. With an optional.

Reply:
I'll try to figure this one out, it's interesting!

Reply:
Let me know if you need help with it. Its actually quite easy. But its feels good to finally figure out a regex :laughing:

Reply:
I figured out the PCRE version of the regex. I though i could just replace `\` by `%` to have it work in lua, but the lua regex engine doesn't seem to allow logical 'or', so I throw the towell haha. Let me know how! The PCRE Regex I ended up with : `([RC].)\s.+\s\-\>\s(.+)|(..)\s(.+)`

Reply:
Okay i wasn't thinking about `or` and i don't even know how to do that in lua.
I ended up doing `'(..).*%s[->%s]?(.+)'`

Reply:
Hahaha this is wizardry ! 😂

Can you please explain to me how it works? 🙏🏿

Reply:
Sure 
- `(..)` = will match the first two chars.
- `.*` = will do anything or nothing. Nothing is important
- `%s` = one white space. Thats the whitespace between `xx file` or `xx oldfile `(last one). why the last one because `.*` would match oldfile.
- `[->%s]?` = will match these characters optional. So this only applies when we have an arrow. for `xx file` we still only matched `xx ` and for `xx oldfile -> newfile` we have matched `xx oldfile -> ` at this point.
- `(.+)` = will just match till the end of the string.